### PR TITLE
Fix automatic game log attachment (Android)

### PR DESCRIPTION
**Pull request by tustanivsky**
Related to #302

### CODE REVIEW

File: plugin-dev/Source/Sentry/Private/Android/Java/SentryBridgeJava.java [hunk 1]
```diff
@@ -45,15 +45,24 @@ public void configure(SentryAndroidOptions options) {
...
-			}
-		});
-
-		Sentry.configureScope(new ScopeCallback() {
-			@Override
-			public void run(@NonNull Scope scope) {
-				if(!gameLogPath.isEmpty()) {
-					scope.addAttachment(new Attachment(gameLogPath));
-				}
+				options.setBeforeSend(new SentryOptions.BeforeSendCallback() {
```
Comment (bitsandfoxes) [inline plugin-dev/Source/Sentry/Private/Android/Java/SentryBridgeJava.java:48]: I'm not entirely sure how that works underneath but can that callback be overwritten by users?

File: plugin-dev/Source/Sentry/Private/Android/Java/SentryBridgeJava.java [hunk 2]
```diff
@@ -45,15 +45,24 @@ public void configure(SentryAndroidOptions options) {
...
+					@Override
+					public SentryEvent execute(SentryEvent event, Hint hint) {
+						if(event.isCrashed() && event.isErrored())
```
Comment (bitsandfoxes) [inline plugin-dev/Source/Sentry/Private/Android/Java/SentryBridgeJava.java:51]: Doesn't that mean that if the app crashed in the run before the gamelogs from the current run won't be part of any current events?

File: plugin-dev/Source/Sentry/Private/Android/Java/SentryBridgeJava.java (hunk 1 above)
Comment (tustanivsky) [reply plugin-dev/Source/Sentry/Private/Android/Java/SentryBridgeJava.java:48]: We don't expose any APIs allowing users to do that, so it should be safe to set the `beforeSend` hook like this.

Comment (bitsandfoxes) [reply plugin-dev/Source/Sentry/Private/Android/Java/SentryBridgeJava.java:48]: Considering that we want to expose that API in the future, we could consider to add this as an eventprocessor/integration instead?

File: plugin-dev/Source/Sentry/Private/Android/Java/SentryBridgeJava.java (hunk 2 above)
Comment (tustanivsky) [reply plugin-dev/Source/Sentry/Private/Android/Java/SentryBridgeJava.java:51]: No, after sending the crash event on app startup all subsequent events should be treated as usual with a current session's game log attached.

File: plugin-dev/Source/Sentry/Private/Android/Java/SentryBridgeJava.java (hunk 1 above)
Comment (tustanivsky) [reply plugin-dev/Source/Sentry/Private/Android/Java/SentryBridgeJava.java:48]: Yeah, I think it should be feasible.

**Pull request merged**